src/Qir/Runtime/lib/QIR/bridge-rt.ll

```diff
@@ -1,284 +1,20 @@
-  %ar = bitcast %Array* %.ar to %"struct.QirArray"*
-  %ar_copy = call %"struct.QirArray"* @quantum__rt__array_copy(%"struct.QirArray"* %ar, i1 %force)
-  %.ar_copy = bitcast %"struct.QirArray"* %ar_copy to %Array*
-  ret %Array* %.ar_copy
-}
-
-; TODO: This bridge isn't cross-platform!
-; it works on Windows but on Linux %args ends up not being a valid pointer.
-define dllexport %Array* @__quantum__rt__array_create(i32 %item_size, i32 %dim_count, ...) {
-  %args1 = alloca i8*, align 8
-  %args2 = bitcast i8** %args1 to i8*
-  call void @llvm.va_start(i8* %args2)
-  %args = load i8*, i8** %args1, align 8
-  %ar = call %"struct.QirArray"* @quantum__rt__array_create_nonvariadic(i32 %item_size, i32 %dim_count, i8* %args)
-  call void @llvm.va_end(i8* %args2)
-  %.ar = bitcast  %"struct.QirArray"* %ar to %Array*
-  ret %Array* %.ar
-}
-
-define dllexport %Array* @__quantum__rt__array_create_1d(i32 %item_size, i64 %count) {
-  %ar = call %"struct.QirArray"* @quantum__rt__array_create_1d(i32 %item_size, i64 %count)
-  %.ar = bitcast  %"struct.QirArray"* %ar to %Array*
-  ret %Array* %.ar
-}
-
-define dllexport i32 @__quantum__rt__array_get_dim(%Array* %.ar) {
-  %ar = bitcast %Array* %.ar to %"struct.QirArray"*
-  %dim_count = call i32 @quantum__rt__array_get_dim(%"struct.QirArray"* %ar)
-  ret i32 %dim_count
-}
-
-define dllexport i8* @__quantum__rt__array_get_element_ptr(%Array* %.ar, ...) {
-  %args1 = alloca i8*, align 8
-  %args2 = bitcast i8** %args1 to i8*
-  call void @llvm.va_start(i8* %args2)
-  %args = load i8*, i8** %args1, align 8
-  %ptr = call i8* @quantum__rt__array_get_element_ptr_nonvariadic(%Array* %.ar, i8* %args)
-  call void @llvm.va_end(i8* %args2)
-  ret i8* %ptr
-}
-
-define dllexport i8* @__quantum__rt__array_get_element_ptr_1d(%Array* %.ar, i64 %i) {
-  %ar = bitcast %Array* %.ar to %"struct.QirArray"*
-  %item =
-    call i8* @quantum__rt__array_get_element_ptr_1d(%"struct.QirArray"* %ar, i64 %i)
-  ret i8* %item
-}
-
-define dllexport i64 @__quantum__rt__array_get_size(%Array* %.ar, i32 %dim) {
-  %ar = bitcast %Array* %.ar to %"struct.QirArray"*
-  %l = call i64 @quantum__rt__array_get_size(%"struct.QirArray"* %ar, i32 %dim)
-  ret i64 %l
-}
-
-define dllexport i64 @__quantum__rt__array_get_size_1d(%Array* %.ar) {
-  %l = call i64 @__quantum__rt__array_get_size(%Array* %.ar, i32 0)
-  ret i64 %l
-}
-
-define dllexport %Array* @__quantum__rt__array_project(%Array* %.ar, i32 %dim, i64 %index) {
-  %ar = bitcast %Array* %.ar to %"struct.QirArray"*
-  %project = call %"struct.QirArray"* @quantum__rt__array_project(%"struct.QirArray"* %ar, i32 %dim, i64 %index)
-  %.project = bitcast  %"struct.QirArray"* %project to %Array*
-  ret %Array* %.project
-}
+; NOTE: These three functions can be converted to extern C once the spec and compiler are updated to pass %Range by
+; pointer instead of by value (see https://github.com/microsoft/qsharp-language/issues/108). Once that
+; happens, this file can be removed. 

```

Comment:
Drawing extra attention to this comment. The only .ll functions left are used to handle `%Range` being passed by value. If we fix the spec and the corresponding code generation in the compiler, then these functions change to something like `define dllexport %String* @__quantum__rt__range_to_string(%Range* %.range)` and can be implemented directly in C++ like the rest of the runtime.